tokio/src/net/unix/pipe.rs

```diff
@@ -623,6 +695,21 @@ impl Sender {
             .registration()
             .try_io(Interest::WRITABLE, || (&*self.io).write_vectored(buf))
     }
+
+    /// Converts the pipe into an [`OwnedFd`].
+    ///
+    /// This function will deregister this pipe end from the event loop, set
+    /// it in blocking mode and perform the conversion.
+    pub fn into_owned_fd(self) -> io::Result<OwnedFd> {
+        let mio_pipe = self.io.into_inner()?;
+        set_blocking(&mio_pipe)?;
```

Comment:
What happens if `set_blocking` fails? Is the fd closed?

Reply:
The file descriptor gets closed once `mio_pipe` is dropped.